components/suggest/src/rs.rs

```diff
@@ -69,85 +72,58 @@ pub(crate) trait Client {
     /// client-side filtering.
     ///
     /// Records that can't be parsed as [SuggestRecord] are ignored.
-    fn get_records(&self, collection: Collection, dao: &mut SuggestDao) -> Result<Vec<Record>>;
+    fn get_records(&self, collection: Collection) -> Result<Vec<Record>>;
 
     fn download_attachment(&self, record: &Record) -> Result<Vec<u8>>;
 }
 
 /// Implements the [Client] trait using a real remote settings client
-pub struct RemoteSettingsClient {
+pub struct SuggestRemoteSettingsClient {
     // Create a separate client for each collection name
-    quicksuggest_client: remote_settings::RemoteSettings,
-    fakespot_client: remote_settings::RemoteSettings,
+    quicksuggest_client: Arc<RemoteSettingsClient>,
+    fakespot_client: Arc<RemoteSettingsClient>,
 }
 
-impl RemoteSettingsClient {
-    pub fn new(
-        server: Option<remote_settings::RemoteSettingsServer>,
-        bucket_name: Option<String>,
-        server_url: Option<String>,
-    ) -> Result<Self> {
+impl SuggestRemoteSettingsClient {
+    pub fn new(rs_service: &RemoteSettingsService) -> Result<Self> {
         Ok(Self {
-            quicksuggest_client: remote_settings::RemoteSettings::new(
-                remote_settings::RemoteSettingsConfig {
-                    server: server.clone(),
-                    bucket_name: bucket_name.clone(),
-                    collection_name: "quicksuggest".to_owned(),
-                    server_url: server_url.clone(),
-                },
-            )?,
-            fakespot_client: remote_settings::RemoteSettings::new(
-                remote_settings::RemoteSettingsConfig {
-                    server,
-                    bucket_name,
-                    collection_name: "fakespot-suggest-products".to_owned(),
-                    server_url,
-                },
-            )?,
+            quicksuggest_client: rs_service.make_client("quicksuggest".to_owned(), None)?,
+            fakespot_client: rs_service
+                .make_client("fakespot-suggest-products".to_owned(), None)?,
         })
     }
 
-    fn client_for_collection(&self, collection: Collection) -> &remote_settings::RemoteSettings {
+    fn client_for_collection(&self, collection: Collection) -> &Arc<RemoteSettingsClient> {
```

Comment:
Nit: this could just be a `&RemoteSettingsClient`.  The `Arc` makes me think it might be cloned to create a new reference, but I don't think you ever do that.